meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClientCache.java

```diff
@@ -106,26 +103,56 @@ public int countDirectChildren(final String key) {
         throw new MetaClientException("Not implemented yet.");
     }
 
+    private void populateAllCache() {
+        if (_cacheData) {
+            try {
+                List<String> children = _cacheClient.getChildren(_rootEntry);
+                for (String child : children) {
+                    String childPath = _rootEntry + "/" + child;
+                    T dataRecord = _cacheClient.readData(childPath, true);
```

Comment:
can we do batch read?

Reply:
Batch read isn't supported in super class at the moment